metadata/src/lib.rs

```diff
@@ -55,11 +55,11 @@ where
 pub trait Metadata: Send + Sized + 'static {
     type Message: protobuf::Message;
 
-    fn base_url() -> &'static str;
+    fn request_url(id: SpotifyId) -> String;
```

Comment:
Is there a benefit to this change? It looks like it just moves more code into each implementation of `Metadata`.

Reply:
This is to allow different types of metadata use different formatting of the endpoint. Playlists use base62 Uris while all others use base16 and that didn't fit well with the existing solution.

Reply:
Ahh I entirely missed that difference. Apologies.